### PR TITLE
Fix null dereference when the LLM head Transpose order is not a Constant

### PR DESCRIPTION
**Pull request by sgonorov**
## Description

Coverity CID 6954501 (`NULL_RETURNS`, CWE-476)

CVS-193071

`find_llm_matmul()` cast the Transpose `order` input to a `Constant` and dereferenced the
result without checking it. A model whose logits end in a Transpose with a computed
(non-constant) order made that cast return null, so the call crashed.

The order is now checked before use. A non-constant order throws instead of crashing, and
the permutation is also bounds-checked before `order_values[slice_gather_dim]` reads it.

Two CI fixes ride along so the job set is green:

- `test_linear_attention_batch_string_inputs` xfail for qwen3-next + PAGED_ATTENTION now
  covers every batch in `BATCHED_PROMPTS`, not just one. All of them have been seen
  diverging (CVS-192310).
- `test_pruner_basic` is xfailed on Linux, where wwb aborts at interpreter shutdown with
  `PyGILState_Release` (ticket 194373).

## Checklist:
- [x] This PR follows [GenAI Contributing guidelines](https://github.com/openvinotoolkit/openvino.genai?tab=contributing-ov-file#contributing).
- [x] Tests have been updated or added to cover the new code. Added `SliceBeforeMatmul.NonConstantTransposeOrderThrows` and `GatherBeforeMatmul.NonConstantTransposeOrderThrows` in `tests/cpp/test_slice_before_matmul.cpp`. Both assert that a non-constant Transpose order raises `ov::Exception` rather than dereferencing null.
- [x] This PR fully addresses the ticket.
- [x] N/A — no documentation changes, the fix is internal to a graph transformation.

### CODE REVIEW

File: src/cpp/src/utils.cpp [hunk 1]
```diff
@@ -334,9 +335,18 @@ std::tuple<std::shared_ptr<ov::Node>, int64_t> find_llm_matmul(const std::shared
         if (auto add = ov::as_type_ptr<ov::op::v1::Add>(last_node)) {
             matmul = ov::as_type_ptr<ov::op::v0::MatMul>(add->input_value(0).get_node_shared_ptr());
         } else if (auto transpose = ov::as_type_ptr<ov::op::v1::Transpose>(last_node)) {
+            auto order = ov::as_type_ptr<ov::op::v0::Constant>(transpose->input_value(1).get_node_shared_ptr());
+            if (!order) {
+                GENAI_WARN("Transpose before the LLM head has a non-constant order, "
+                           "skipping slice / gather before MatMul.");
```
Comment (Wovchena) [inline src/cpp/src/utils.cpp]: Make it an error. null `order` before the patch resulted an error so you'll preserve the behaviour

Comment (Wovchena) [reply src/cpp/src/utils.cpp]: The error is still missing

Comment (sgonorov) [reply src/cpp/src/utils.cpp]: Fixed

**Review by Wovchena — APPROVED** [review]
